pkg/providers/cloudstack/cloudstack.go

```diff
@@ -657,10 +659,34 @@ func (cs *CloudStackTemplateBuilder) GenerateCAPISpecControlPlane(clusterSpec *c
 		buildOption(values)
 	}
 
+	cpCsMtName, ok := values[cpTemplateNameKey]
+	if !ok {
+		return nil, fmt.Errorf("unable to determine control plane template name")
+	}
+	cpMachineTemplate := CloudStackMachineTemplate(fmt.Sprintf("%v", cpCsMtName), cs.controlPlaneMachineSpec, cs.now)
```

Comment:
Just so I can see how this will fit the big picture, how are you planning on using this from the controller?

Are the code paths gonna be the same for cli and controller (minus the marshaling) or are they going to be composed differently

Reply:
all the immutable objects are going to be generated from the apibuilder (machinetemplates and machinedeployments) in both the CLI and the controller. All the mutable objects will be generated from the templates (kubeadmconfig, cloudstackcluster, capi cluster) in both the CLI and controller. In the CLI, the objects will be applied as yaml. In the controller, they will be compared and then applied as K8s objects.

Reply:
I see. But we are in `GenerateCAPISpecControlPlane` method, which returns `[]byte`
Then does this mean we are going to need to duplicate the logic that calls "go template + api structs" in the controller? Just to be able to access the api structs before they get marshaled to yaml